recipes/CodeLLM_DevKit/code_summarization.ipynb

```diff
@@ -0,0 +1,426 @@
+    "\n",
+    "    inst += \"\\n\"\n",
+    "    inst += f\"```{language}\\n\"\n",
+    "    inst += code\n",
+    "    inst += \"```\" if code.endswith(\"\\n\") else \"\\n```\"\n",
+    "    inst += \"\\n\"\n",
+    "    return inst"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a4e9cb4e4f00b25c",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dd8439be222b5caa",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "Step 3: Define a function to call the LLM (in this case, Granite code 8b-instruct) using Ollama."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "62807e0cbf985ae6",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "def prompt_ollama(message: str, model_id: str = \"granite-code:8b-instruct\") -> str:\n",
+    "    \"\"\"Prompt local model on Ollama\"\"\"\n",
+    "    response_object = ollama.generate(model=model_id, prompt=message, options={\"temperature\":0.2})\n",
+    "    return response_object[\"response\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1022e86e38e12767",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "Step 4: Create an instance of CLDK and provide the programming language of the source code."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a2c8bbe4e3244f60",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Create an instance of CLDK for Java analysis\n",
+    "cldk = CLDK(language=\"java\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "23dd4a6e5d5cb0c5",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "Step 5: Select the analysis engine and analysis level. CLDK uses different analysis engines---[CodeAnalyzer](https://github.com/IBM/codenet-minerva-code-analyzer) (built over [WALA](https://github.com/wala/WALA) and [JavaParser](https://github.com/javaparser/javaparser)), [Treesitter](https://tree-sitter.github.io/tree-sitter/), and [CodeQL](https://codeql.github.com/) (future)---with CodeAnalyzer being the default analysis engine. CLDK supports different analysis levels: (1) symbol table, (2) call graph, (3) program dependency graph, and (4) system dependency graph. The analysis level can be selected using the `AnalysisLevel` enumerated type. For this example, we select the symbol-table analysis level, with CodeAnalyzer as the default analysis engine."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fdd09f5e77d4a68a",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Create an analysis object for the Java application\n",
+    "analysis = cldk.analysis(project_path=\"/tmp/commons-cli-rel-commons-cli-1.7.0\", analysis_level=AnalysisLevel.symbol_table)"
```

Comment:
This fails with:
```
CodeanalyzerExecutionException: Command '['java', '-jar', '/Users/fayvor/Dev/granite-code-cookbook/.venv/lib/python3.12/site-packages/cldk/analysis/java/codeanalyzer/jar/codeanalyzer.20240830T123519.jar', '-i', '/tmp/commons-cli-rel-commons-cli-1.7.0', '--analysis-level=1', '-o', '.']' returned non-zero exit status 1.
```

When I run that java command at the cli, I see that I don't have `mvn`. Maybe add maven to the dependencies above?

Reply:
Added maven as a prerequisite